### PR TITLE
update tests from core to run together

### PR DESCRIPTION
**Pull request by vhvb1989**
Fixing test cases for Context and Retry
- Context was calling Cancel to the global context, making any next test trying to use the global context to fail
- Retry policy tests uses a global state for counting which needs to be reset for each test

### CODE REVIEW

File: sdk/core/azure-core/test/ut/context.cpp
Comment (ahsonkhan) [inline sdk/core/azure-core/test/ut/context.cpp:86]: Looks like the intention of this test was specifically to test `ApplicationContext` from the test name, which we are no longer testing.

How are we supposed to test `ApplicationContext` (along with cancelling it)?

Maybe all other tests should avoid using a global context. To keep the test at least somewhat related to ApplicationContext, change `appContext` to be a child of the root ApplicationContext.

File: sdk/core/azure-core/test/ut/policy.cpp
Comment (ahsonkhan) [inline sdk/core/azure-core/test/ut/policy.cpp:160]: Good idea.

File: sdk/core/azure-core/test/ut/context.cpp
Comment (vhvb1989) [reply sdk/core/azure-core/test/ut/context.cpp:86]: Application context is not different than any Context.
It would be like testing a global std::string instead of any std::string specifically.

We call GetAppContext for multiple places.  If we really want to have such a test (useless), we would want to create a unique test exe just for that test so it never affect the other tests.

**Review by RickWinter — CHANGES_REQUESTED** [review]

**Review by ahsonkhan — APPROVED** [review]
Wait for review/approval after you address the merge conflict. Looks good so far.

File: sdk/core/azure-core/test/ut/policy.cpp
Comment (ahsonkhan) [inline sdk/core/azure-core/test/ut/policy.cpp:174]:
```suggestion
  using namespace Azure::Core::Http::Internal;
```

**Review by ahsonkhan — APPROVED** [review]
Remove extra semi-colons. Looks good otherwise.

**Pull request merged**
